src/System.CommandLine/Binding/ArgumentConverter.cs

```diff
@@ -195,7 +195,7 @@ static IList CreateArray(Type itemType, int capacity)
                       .GetInterfaces()
                       .FirstOrDefault(IsEnumerable);
 
-            return enumerableInterface?.GenericTypeArguments[0];
+            return enumerableInterface?.GenericTypeArguments.FirstOrDefault();
```

Comment:
Though I like the change, I'd revert this. This is one of these little perf optimizations that do add up.

Reply:
I assume you mean replace `.FirstOrDefault()` with a bounds check? Because the `[0]` throws an out of range exception when working with non-generic. `IsEnumerable` returns true for non generic `IEnumerable` so there may not be any generic arguments.

Reply:
Did you hit that exception? At least before, that wasn't happening.

Agreed, this change is safer, but we should attempt to keep the optimization.

Reply:
Yea the exception occurs on the tests I added for non-generic IEnumerable, ICollection, and IList